### PR TITLE
Removes unreferenced, unused bits of code

### PR DESCRIPTION
**Pull request by randall**

### CODE REVIEW

**Comment by randall** [comment]
[approve ci]

**Comment by zwoop** [comment]
merge conflicts.

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -3909,29 +3904,6 @@ HttpSM::tunnel_handler_plugin_agent(int event, HttpTunnelConsumer *c)
...
-HttpSM::state_srv_lookup(int event, void *data)
-{
-  STATE_ENTER(&HttpSM::state_srv_lookup, event);
-
-  ink_assert(t_state.req_flavor == HttpTransact::REQ_FLAVOR_SCHEDULED_UPDATE ||
-             t_state.req_flavor == HttpTransact::REQ_FLAVOR_REVPROXY || ua_entry->vc != nullptr);
-
-  switch (event) {
-  case EVENT_SRV_LOOKUP:
-    pending_action = nullptr;
```
Comment (zwoop) [inline proxy/http/HttpSM.cc]: Did we remove some other code that used this? Or is there new code dealing with these events?

File: proxy/http/HttpTransact.cc [hunk 2]
```diff
@@ -8790,12 +8678,3 @@ HttpTransact::change_response_header_because_of_range_request(State *s, HTTPHdr
     header->set_content_length(s->range_output_cl);
   }
 }
-
-#if TS_HAS_TESTS
-void forceLinkRegressionHttpTransact();
-void
-forceLinkRegressionHttpTransactCaller()
-{
```
Comment (zwoop) [inline proxy/http/HttpTransact.cc]: Is this all dead code?

File: proxy/shared/DiagsConfig.cc [hunk 3]
```diff
@@ -244,26 +244,6 @@ DiagsConfig::config_diags_norecords()
...
-  RecRegisterConfigString(RECT_CONFIG, "proxy.config.diags.debug.tags", "", RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigInt(RECT_CONFIG, "proxy.config.diags.action.enabled", 0, RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigString(RECT_CONFIG, "proxy.config.diags.action.tags", "", RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigInt(RECT_CONFIG, "proxy.config.diags.show_location", 1, RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigString(RECT_CONFIG, "proxy.config.diags.output.diag", "L", RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigString(RECT_CONFIG, "proxy.config.diags.output.debug", "L", RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigString(RECT_CONFIG, "proxy.config.diags.output.status", "L", RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigString(RECT_CONFIG, "proxy.config.diags.output.note", "L", RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
-  RecRegisterConfigString(RECT_CONFIG, "proxy.config.diags.output.warning", "L", RECU_NULL, RECC_NULL, nullptr, REC_SOURCE_DEFAULT);
```
Comment (zwoop) [inline proxy/shared/DiagsConfig.cc]: Do these register in RecordsConfig.cc?

File: proxy/http/HttpSM.cc (hunk 1 above)
Comment (shinrich) [reply proxy/http/HttpSM.cc]: My guess is that logic moved into HttpSM::state_hostdb_lookup.  The HostDB will be calling back to the HttpSM contination with the SRV_LOOKUP event and state_hostdb_lookup is still being set as a handler.

File: proxy/shared/DiagsConfig.cc (hunk 3 above)
Comment (shinrich) [reply proxy/shared/DiagsConfig.cc]: Yes.

**Comment by shinrich** [comment]
Looks like good tidying up to me.  Just need to rebase.

File: include/tscore/ink_string.h [hunk 4]
```diff
@@ -154,7 +154,7 @@ ptr_len_str(const char *p1, int l1, const char *str)
   return nullptr;
 }
 
-// int ptr_len_ncmp(const char* p1, int l1, const char* str, int n) {
+// int ptr_len_ncmp(const char* p1, int l1, const char* str, int n)
 //
```
[the remark this answers is not in the document]
Comment (randall) [reply include/tscore/ink_string.h]: reverted.

File: proxy/http/HttpTransact.cc (hunk 2 above)
Comment (randall) [reply proxy/http/HttpTransact.cc]: Restored.

File: proxy/shared/DiagsConfig.cc (hunk 3 above)
Comment (randall) [reply proxy/shared/DiagsConfig.cc]: I couldn't find anywhere that called this function.

**Comment by randall** [comment]
[approve ci clang-analyzer]

**Comment by randall** [comment]
[approve ci autest]

**Comment by randall** [comment]
Change rebased.

**Comment by randall** [comment]
[approve ci autest]

**Comment by randall** [comment]
@zwoop @shinrich Can you take a look at this now?

**Comment by randall** [comment]
@zwoop @shinrich Can you take a look at this now?

**Pull request merged**